plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -181,6 +186,13 @@ public IcebergSplitSource(
 
             IcebergSplit icebergSplit = toIcebergSplit(scanTask);
 
+            Set<Integer> identityPartitionFieldIds = getIdentityPartitions(scanTask.spec()).keySet().stream()
+                    .map(PartitionField::sourceId)
+                    .collect(toImmutableSet());
+            Set<IcebergColumnHandle> identityPartitionColumns = getColumns(scanTask.spec().schema(), typeManager).stream()
```

Comment:
> scanTask.spec().schema()

this would get the whole schema right? if we're doing this per split, would be worth only iterating over partition columns somehow ... may be use `select` or only create handles for ids we're interested in. wdyt?